smart_card_connector_app/src/reader-tracker.js

```diff
@@ -0,0 +1,212 @@
+/** @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This file contains the class used to track readers and their
+ * status based on messages received from the hook setup on PC/SC code inside
+ * NaCl (check readerfactory_nacl.cc for more information).
+ */
+
+goog.provide('GoogleSmartCard.ReaderTracker');
+
+goog.require('GoogleSmartCard.DebugDump');
+goog.require('GoogleSmartCard.Logging');
+goog.require('GoogleSmartCard.TypedMessage');
+goog.require('goog.asserts');
+goog.require('goog.log.Logger');
+goog.require('goog.messaging.AbstractChannel');
+goog.require('goog.object');
+goog.require('goog.structs.Map');
+
+goog.scope(function() {
+
+/** @const */
+var READER_TRACKER_LOGGER_TITLE = 'ReaderTracker';
+
+/** @const */
+var GSC = GoogleSmartCard;
+
+/**
+ * Enum for possible values of ReaderInfo status.
+ * @enum {string}
+ */
+var ReaderStatus = {
+  INIT: 'init',
+  SUCCESS: 'success',
+  FAILURE: 'failure'
+};
+
+/**
+ * Structure used to store information about the reader.
+ * @param {string} name
+ * @param {ReaderStatus} status
+ * @param {string=} opt_error
+ * @constructor
+ */
+var ReaderInfo = function(name, status, opt_error) {
```

Comment:
Are you not exporting this symbol intentionally?
It's a bit strange to keep this definition "private", given that values of this type are returned through public methods.